cmake/modules/BuildRocksDB.cmake

```diff
@@ -44,6 +44,8 @@ function(build_rocksdb)
   if(HAS_WARNING_STRINGOP_TRUNCATION)
     list(APPEND rocksdb_CMAKE_ARGS -DCMAKE_C_FLAGS="-Wno-stringop-truncation")
   endif()
+  list(APPEND rocksdb_CMAKE_ARGS -DCMAKE_CXX_FLAGS="-Wno-deprecated-copy")
+
```

Comment:
@adamemerson 
My Clang builds also contain quite a few of the warnings below.
Perhaps a -Wno-shadow can suppress them?

```
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:111:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WAL_FILE_BYTES,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:190:3: note: previous declaration is here
  WAL_FILE_BYTES,   // Number of bytes written to WAL
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:112:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WAL_FILE_SYNCED,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:189:3: note: previous declaration is here
  WAL_FILE_SYNCED,  // Number of times WAL sync is done
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:113:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    BYTES_WRITTEN,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:126:3: note: previous declaration is here
  BYTES_WRITTEN,
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:114:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    NUMBER_KEYS_WRITTEN,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:119:3: note: previous declaration is here
  NUMBER_KEYS_WRITTEN,
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:115:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WRITE_DONE_BY_OTHER,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:195:3: note: previous declaration is here
  WRITE_DONE_BY_OTHER,  // Equivalent to writes done for others
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:116:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WRITE_DONE_BY_SELF,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:194:3: note: previous declaration is here
  WRITE_DONE_BY_SELF,
  ^
In file included from /home/jenkins/workspace/ceph-master/src/rocksdb/db/builder.cc:19:
/home/jenkins/workspace/ceph-master/src/rocksdb/db/internal_stats.h:117:5: warning: declaration shadows a variable in namespace 'rocksdb' [-Wshadow]
    WRITE_WITH_WAL,
    ^
/home/jenkins/workspace/ceph-master/src/rocksdb/include/rocksdb/statistics.h:197:3: note: previous declaration is here
  WRITE_WITH_WAL,       // Number of Write calls that request WAL
  ^
7 warnings generated.
```